# III. How do offered salaries compare to the prevailing wage?

## 3a: Are there job sub-categories that tend to get over-paid or under-paid?
## 3b: Are there companies that tend to over-pay or under-pay?


In [21]:
#Percentage of Underpaying Jobs
((7+547+77+10+4+2+307+101)/167278)*100

0.6306866413993472

In [44]:
#Percentage of Overpaying Jobs
((1192+16343+20713+3089+1038+649+71103+7758)/167278)*100

72.86373581702316

### Attorney

#### 3a

In [ ]:
#Underpay
SELECT employer_name,work_state, job_title_subgroup, paid_wage_per_year, prevailing_wage_per_year, paid_wage_per_year-prevailing_wage_per_year AS Diff
FROM salary_data_states_corrected
WHERE job_title_subgroup = "attorney"
ORDER BY paid_wage_per_year-prevailing_wage_per_year ASC
LIMIT 10
##7

In [ ]:
SELECT COUNT(job_title_subgroup)
FROM salary_data_states_corrected
WHERE job_title_subgroup = 'attorney'
##1488

In [19]:
(7/1488)*100

0.47043010752688175

In [ ]:
#Overpay
SELECT COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "attorney" AND paid_wage_per_year-prevailing_wage_per_year>0

In [36]:
(1192/1488)*100

80.10752688172043

#### 3b

In [ ]:
#Companies tending to overpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "attorney" AND paid_wage_per_year-prevailing_wage_per_year>0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_at_overpay.png)

In [ ]:
#Companies tending to underpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "attorney" AND paid_wage_per_year-prevailing_wage_per_year<0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_at_underpay.png)

### Assistant Professor

#### 3a

In [ ]:
#Underpay
SELECT employer_name,work_state, job_title_subgroup, paid_wage_per_year, prevailing_wage_per_year, paid_wage_per_year-prevailing_wage_per_year AS Diff
FROM salary_data_states_corrected
WHERE job_title_subgroup = "assistant professor"
ORDER BY paid_wage_per_year-prevailing_wage_per_year ASC
LIMIT 560
##547

In [ ]:
SELECT COUNT(job_title_subgroup)
FROM salary_data_states_corrected
WHERE job_title_subgroup = 'assistant professor'
##18866

In [18]:
(547/18866)*100

2.8993957383653135

In [ ]:
#Overpay
SELECT COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "assistant professor" AND paid_wage_per_year-prevailing_wage_per_year>0

In [37]:
(16343/18866)*100

86.62673592706456

#### 3b

In [ ]:
#Companies tending to overpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "assistant professor" AND paid_wage_per_year-prevailing_wage_per_year>0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_ap_overpay.png)

In [ ]:
#Companies tending to underpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "assistant professor" AND paid_wage_per_year-prevailing_wage_per_year<0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_ap_underpay.png)

### Business Analyst

#### 3a

In [ ]:
#Underpay
SELECT employer_name,work_state, job_title_subgroup, paid_wage_per_year, prevailing_wage_per_year, paid_wage_per_year-prevailing_wage_per_year AS Diff
FROM salary_data_states_corrected
WHERE job_title_subgroup = "business analyst"
ORDER BY paid_wage_per_year-prevailing_wage_per_year ASC
LIMIT 80
##77

In [ ]:
SELECT COUNT(job_title_subgroup)
FROM salary_data_states_corrected
WHERE job_title_subgroup = 'business analyst'
##27811

In [17]:
(77/27811)*100

0.27686886483765416

In [ ]:
#Overpay
SELECT COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "business analyst" AND paid_wage_per_year-prevailing_wage_per_year>0

In [38]:
(20713/27811)*100

74.47772464132896

#### 3b

In [ ]:
#Companies tending to overpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "business analyst" AND paid_wage_per_year-prevailing_wage_per_year>0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_ba_overpay.png)

In [ ]:
#Companies tending to underpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "business analyst" AND paid_wage_per_year-prevailing_wage_per_year<0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_ba_underpay.png)

### Data Analyst

#### 3a

In [ ]:
#Underpay
SELECT employer_name,work_state, job_title_subgroup, paid_wage_per_year, prevailing_wage_per_year, paid_wage_per_year-prevailing_wage_per_year AS Diff
FROM salary_data_states_corrected
WHERE job_title_subgroup = "data analyst"
ORDER BY paid_wage_per_year-prevailing_wage_per_year ASC
LIMIT 10
##10

In [ ]:
SELECT COUNT(job_title_subgroup)
FROM salary_data_states_corrected
WHERE job_title_subgroup = 'data analyst'
##3840

In [16]:
(10/3840)*100

0.26041666666666663

In [ ]:
#Overpay
SELECT COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "data analyst" AND paid_wage_per_year-prevailing_wage_per_year>0

In [39]:
(3089/3840)*100

80.44270833333333

#### 3b

In [ ]:
#Companies tending to overpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "data analyst" AND paid_wage_per_year-prevailing_wage_per_year>0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_da_overpay.png)

In [ ]:
#Companies tending to underpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "data analyst" AND paid_wage_per_year-prevailing_wage_per_year<0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_da_underpay.png)

### Data Scientist

#### 3a

In [ ]:
#Underpay
SELECT employer_name,work_state, job_title_subgroup, paid_wage_per_year, prevailing_wage_per_year, paid_wage_per_year-prevailing_wage_per_year AS Diff
FROM salary_data_states_corrected
WHERE job_title_subgroup = "data scientist"
ORDER BY paid_wage_per_year-prevailing_wage_per_year ASC
LIMIT 5
##4

In [ ]:
SELECT COUNT(job_title_subgroup)
FROM salary_data_states_corrected
WHERE job_title_subgroup = 'data scientist'
##1227

In [15]:
(4/1227)*100

0.32599837000814996

In [ ]:
#Overpay
SELECT COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "data scientist" AND paid_wage_per_year-prevailing_wage_per_year>0

In [40]:
(1038/1227)*100

84.59657701711491

#### 3b

In [ ]:
#Companies tending to overpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "data scientist" AND paid_wage_per_year-prevailing_wage_per_year>0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_ds_overpay.png)

In [27]:
#Companies tending to underpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "data scientist" AND paid_wage_per_year-prevailing_wage_per_year<0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_ds_underpay.png)

### Management Consultant

#### 3a

In [ ]:
#Underpay
SELECT employer_name,work_state, job_title_subgroup, paid_wage_per_year, prevailing_wage_per_year, paid_wage_per_year-prevailing_wage_per_year AS Diff
FROM salary_data_states_corrected
WHERE job_title_subgroup = "management consultant"
ORDER BY paid_wage_per_year-prevailing_wage_per_year ASC
LIMIT 5
##2

In [ ]:
SELECT COUNT(job_title_subgroup)
FROM salary_data_states_corrected
WHERE job_title_subgroup = 'management consultant'
##770

In [14]:
(2/770)*100

0.2597402597402597

In [ ]:
#Overpay
SELECT COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "management consultant" AND paid_wage_per_year-prevailing_wage_per_year>0

In [41]:
(649/770)*100

84.28571428571429

#### 3b

In [28]:
#Companies tending to overpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "management consultant" AND paid_wage_per_year-prevailing_wage_per_year>0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

SyntaxError: invalid syntax (23924318.py, line 2)

![t_overpay](images/q3_m_overpay.png)

In [ ]:
#Companies tending to underpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "management consultant" AND paid_wage_per_year-prevailing_wage_per_year<0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_m_underpay.png)

### Software Engineer

#### 3a

In [ ]:
#Underpay
SELECT employer_name,work_state, job_title_subgroup, paid_wage_per_year, prevailing_wage_per_year, paid_wage_per_year-prevailing_wage_per_year AS Diff
FROM salary_data_states_corrected
WHERE job_title_subgroup = "software engineer"
ORDER BY paid_wage_per_year-prevailing_wage_per_year ASC
LIMIT 310
##307

In [ ]:
SELECT COUNT(job_title_subgroup)
FROM salary_data_states_corrected
WHERE job_title_subgroup = 'software engineer'
##99364

In [12]:
(307/99364)*100

0.30896501751137234

In [29]:
#Overpay
SELECT COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "software engineer" AND paid_wage_per_year-prevailing_wage_per_year>0

99.69103498248863

In [42]:
(71103/99364)*100

71.55810957690915

#### 3b

In [ ]:
#Companies tending to overpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "software engineer" AND paid_wage_per_year-prevailing_wage_per_year>0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_s_overpay.png)

In [ ]:
#Companies tending to underpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "software engineer" AND paid_wage_per_year-prevailing_wage_per_year<0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_s_underpay.png)

### Teacher

#### 3a

In [ ]:
#Underpay
SELECT employer_name,work_state, job_title_subgroup, paid_wage_per_year, prevailing_wage_per_year, paid_wage_per_year-prevailing_wage_per_year AS Diff
FROM salary_data_states_corrected
WHERE job_title_subgroup = "teacher"
ORDER BY paid_wage_per_year-prevailing_wage_per_year ASC
LIMIT 105
##101

In [ ]:
SELECT COUNT(job_title_subgroup)
FROM salary_data_states_corrected
WHERE job_title_subgroup = 'teacher'
##13912

In [13]:
(101/13912)*100

0.7259919493962047

In [ ]:
#Overpay
SELECT COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "teacher" AND paid_wage_per_year-prevailing_wage_per_year>0

In [43]:
(7758/13912)*100

55.76480736055204

#### 3b

In [ ]:
#Companies tending to overpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "teacher" AND paid_wage_per_year-prevailing_wage_per_year>0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_overpay](images/q3_t_overpay.png)

In [ ]:
#Companies tending to underpay
SELECT employer_name,work_state,paid_wage_per_year-prevailing_wage_per_year AS Diff,COUNT(paid_wage_per_year-prevailing_wage_per_year)
FROM salary_data_states_corrected
WHERE job_title_subgroup = "teacher" AND paid_wage_per_year-prevailing_wage_per_year<0
GROUP BY employer_name
ORDER BY COUNT(paid_wage_per_year-prevailing_wage_per_year) DESC
LIMIT 5

![t_underpay](images/q3_t_underpay.png)

## Interpretation

### What does this all reveal?

#### 3a: Are there job sub-categories that tend to get over-paid or under-paid?
Well overall, 0.63% of all jobs underpay their workers while 72.86% overpay their workers. (The difference between paid wage per year and prevailing wage per year being negative indicates underpaying, while it being positive AND above zero indicates overpaying workers).

Within the assistant professor job subgroup, 2.90% of jobs underpay their workers, and 86.63% of jobs overpay.

Within the teacher job subgroup, 0.73% of jobs underpay their workers, and 55.76% of jobs overpay. 

Within the attorney job subgroup, 0.47% of jobs underpay their workers, and 80.11% of jobs overpay.

Within the data scientist job subgroup, 0.33% of jobs underpay their workers, and 84.60% of jobs overpay.

Within the software engineer job subgroup, 0.31% of jobs underpay their workers, and 71.56% of jobs overpay.

Within the business analyst job subgroup, 0.28 % of jobs underpay their workers, and 74.48% of jobs overpay.

Within the data analyst job subgroup, 0.26% of jobs underpay their workers, and 80.44% of jobs overpay.

Within the management consultant job subgroup, 0.26% of jobs underpay their workers, and 84.29% of jobs overpay.
    
#### 3b: Are there companies that tend to over-pay or under-pay?
The top 5 companies that overpay assistant professors are the Univeristy of Illinois, Purdue Univeristy, the University of Arizona, Pennsylvania State University, and the Univeristy of Michigan.
The top 5 companies that underpay assistant professors are the Univeristy of Tennessee Health Science Center, the University of Alabama, the Univeristy of Virginia Health Services Foundation, and Purdue University.

The top 5 companies that overpay attorneys are Skadden, Arps, Slate, Meagher & Flom LLP, Simpson Thacher & Bartlett LLP, Shearman & Sterling LLP, White & Case LLP, and Sidely Austin (US) LLP.
The top 5 companies that underpay attorneys are DRC Emergency Services, LLC, McCormack & Mattei, P.C., Intelisip Networks, LLC. Sidley Austin (US) LLC, and Georgia Law Center on Homelessness and Poverty.

The top 5 companies that overpay business analysts are Deloitte Consulting LLP, Accenture LLP, SCM Data, Inc., Igate Technollogies Inc., and Hitachi Consulting Corporation.
The top 5 companies that underpay business analysts are Akvarr Inc., Itech Us, Inc., Ana-Data Consulting Inc., Radus Tek Services, Inc., and Vigiboss, Inc..

The top 5 companies that overpay data analysts are Capital One Services, LLC, Ulab Inc., Bell Info Solutions, LLC., Comscore, zinc., and Hanusoft Inc..
The top 5 companies that underpay data analysts are Zoosk, Inc., Capgemini U.S. LLC, Inovalon, Inc., Square, Inc., and Genesis Research LLC.

The top 5 companies that overpay data scientists are Micorsoft Corporation, Facebook, Inc., Linkedin Corporation, Twitter, Inc., and Groupon, Inc..
The top 5 companies that underpay data scientists are Hi-Tech Solutions, Inc., Groupon, Inc., EMC Corporation, and the Advanced Information Management Technology Partner.

The top 5 companies that overpay management consultants are McKinsey & Company, Inc. United States, IBM India Private Limited, IBM Corporation, Dell USA L.P., and Gopa IT Consultants Inc..
The top 5 companies that underpay management consultants are Joe S. Fish Market Inc. and Global Security Solution USA Inc..

The top 5 companies that overpay software engineers are Google Inc., Micorsoft Corporation, Facebook, Inc., Cerner Corporation, and VMWare, Inc.
The top 5 companies that underpay software engineers are Sudhi Informatics Inc., IBM India Private Limited, IBM Corporation, Google Inc., and the Software Paradigms International Group, LLC.

The top 5 companies that overpay teachers are Dallas Independent School District, Houston Independent School District, Global Teachers Research & Resources, Inc., Fort Worth Independent School District, and the Garland Independent School District.
The top 5 companies that underpay teachers are Global Teachers Research & Resources, Inc., Teaching House Inc., Progreso Independent School District, Kodomono Kuni, and the Forest Hill Public Schools. 